docs/docs/recognizers/strategy.md

```diff
@@ -0,0 +1,50 @@
+# Strategy
+## Introduction
+
+### Context of use
+The strategy pattern comes in handy when multiple methods do the same thing in different ways, this pattern is in this case used to reduce code duplication. The gang of four describes the Strategy Pattern in the following manner: “Define a family of algorithms, encapsulate each one, and make them interchangeable. Strategy lets the algorithm vary independently from clients that use it”. It enables the user to use different business rules or algorithms depending on the context in which they occur.
+
+### Example of use
+An example case that shows the need of the pattern is when someone uses a route calculator which calculates the shortest route from A to B. Someone then might want to use the strategy pattern for each of the different routing options (Bike, Car, etc). Then the code can just call a function CreateRoute independent of which routing algorithm is selected.
+
+## Explanation of Architecture
+![UML Strategy](https://refactoring.guru/images/patterns/diagrams/strategy/structure.png)
+
+The **Context** is the class from where the code gets executed. It maintains a reference to one of the concrete strategies and communicates with the object only via the Strategy interface.
+
+The **Strategy** is an interface. This interface needs to contain a method that will be used to execute the algorithm as defined in the ConcreteStrategie classes.
+
+The **ConcreteStrategy** class is designed for one specific purpose and implements the interface Strategy. This class defines the execute() function and implements the algorithm in its own way.
+
+The **Client** creates a ConcreteStrategy object and passes it to the Context. The Context exposes a setter setStrategy() which lets clients replace the strategy associated with the context at runtime.
+
+## Requirements
+_Client class_
+a. has created an object of the type ConcreteStrategy
+b. has used the setStrategy() in the Context class to store the ConcreteStrategy object	
+c. has executed the ConcreteStrategy via the Context class	
+
+_Context class_
+a. has a private field or property that has the Strategy class as type 
+b. has a function setStrategy() to set the private field / property with a parameter of type Strategy
+c. has a function useStrategy() to execute the strategy.
+
+_Strategy interface_
+a. is an interface / abstract class	(duplicate if 4.d holds and the implementation / inheritance is correct)
```

Comment:
In line 15 you say it needs to be an interface

Reply:
I changed this, see above